library/std/src/sys/thread_local/native/eager.rs

```diff
@@ -30,16 +30,22 @@ impl<T> Storage<T> {
     /// The `self` reference must remain valid until the TLS destructor is run.
     #[inline]
     pub unsafe fn get(&self) -> *const T {
-        match self.state.get() {
-            State::Alive => self.val.get(),
-            State::Destroyed => ptr::null(),
-            State::Initial => unsafe { self.initialize() },
+        if let State::Alive = self.state.get() {
+            self.val.get()
+        } else {
+            unsafe { self.get_or_init_slow() }
```

Comment:
I don't think this is beneficial – the returned pointer is later compared against null in `LocalKey::with` anyway, so the optimiser should be able to merge the state comparison into that.